src/Microsoft.DotNet.Interactive.Tests/ImportNotebookTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.DotNet.Interactive.CSharp;
+using Microsoft.DotNet.Interactive.Documents;
+using Microsoft.DotNet.Interactive.Documents.Jupyter;
+using Microsoft.DotNet.Interactive.Events;
+using Microsoft.DotNet.Interactive.FSharp;
+using Microsoft.DotNet.Interactive.Tests.Utility;
+using Xunit;
+
+namespace Microsoft.DotNet.Interactive.Tests
+{
+    public class ImportNotebookTests
+    {
+        [Theory]
+        [InlineData(".ipynb")]
+        [InlineData(".dib")]
+        public async Task It_imports_and_runs(string notebookExt)
+        {
+            using var kernel = new CompositeKernel { 
+                new CSharpKernel(),
+                new FSharpKernel(),
+                new HtmlKernel()
+            }
+                .UseImportMagicCommand();
+
+            var document = new InteractiveDocument
+            {
+                new InteractiveDocumentElement
+                {
+                    Contents = "6+5",
+                    KernelName = "csharp"
+                },
+                new InteractiveDocumentElement
+                {
+                    Contents = "5+3",
+                    KernelName = "markdown" //should not evaluate to 8
+                },
+                new InteractiveDocumentElement
+                {
+                    Contents = "5+3",
```

Comment:
the case for html and markdown is different as they won't produce that event. Html will handle the code submission. very likely you will see the events for that command.